src/Microsoft.VisualStudio.ProjectSystem.Managed/Threading/Tasks/CancellationSeries.cs

```diff
@@ -29,23 +30,30 @@ public CancellationSeries(CancellationToken token = default)
             _superToken = token;
         }
 
+#if DEBUG
+        ~CancellationSeries()
+        {
+            Debug.Assert(_cts == null, "Instance of CancellationSeries not disposed before being finalized");
```

Comment:
In Roslyn we had a lot of trouble with things like this in unit tests due to shutdown.   We ended up wrapping most of them in `if (!Environment.ShutdownStarted)`.

Reply:
Also, probably doesn't matter since this is debug only, but consider adding a `SupppressFinalize` to the `Dispose` method.

Reply:
Thanks, fixed both. Also noticed I didn't protect against exceptions from `Cancel` in `Dispose` so fixed that too.